sklearn/ensemble/voting_classifier.py

```diff
@@ -0,0 +1,192 @@
+"""
+Soft Voting/Majority Rule classifier
+
+This module contains a Soft Voting/Majority Rule classifier for 
+classification estimators.
+
+"""
+
+# Authors: Sebastian Raschka <se.raschka@gmail.com>
+#
+# Licence: BSD 3 clause
+
+from ..base import BaseEstimator
+from ..base import ClassifierMixin
+from ..base import TransformerMixin
+from ..preprocessing import LabelEncoder
+import numpy as np
+import operator
+
+class VotingClassifier(BaseEstimator, ClassifierMixin, TransformerMixin):
+    """ 
+    Soft Voting/Majority Rule classifier for classification estimators.
+        
+    Parameters
+    ----------
+    estimators : array-like, shape = [n_classifiers]
+      A list of estimators for classification.
+      
+    voting : str, {'hard', 'soft'} (default='hard')
+      If 'hard', uses predicted class labels for majority rule voting.
+      Else if 'soft', predicts the class label based on the argmax of 
+      the sums of predict probalities.
+    
+    weights : array-like, shape = [n_classifiers], optional (default=`None`)
+      Sequence of weights (`float` or `int`) that are multiplied with the 
+      predicted class probabilities before averaging if `voting='soft'` 
+      Uses uniform weights if `None`.
+        
+    Attributes
+    ----------
+    classes_ : array-like, shape = [n_predictions]
+        
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.linear_model import LogisticRegression
+    >>> from sklearn.naive_bayes import GaussianNB 
+    >>> from sklearn.ensemble import RandomForestClassifier
+    >>> np.random.seed(123)
+    >>> clf1 = LogisticRegression()
+    >>> clf2 = RandomForestClassifier()
+    >>> clf3 = GaussianNB()
+    >>> X = np.array([[-1, -1], [-2, -1], [-3, -2], [1, 1], [2, 1], [3, 2]])
+    >>> y = np.array([1, 1, 1, 2, 2, 2])
+    >>> eclf1 = VotingClassifier(estimators=[clf1, clf2, clf3], voting='hard')
```

Comment:
It would be nice to have an example where the outputs differ between the options and where the user can understand why they do. That might be tricky, but at least for the weights it might be possible.